installer/create-database.php

```diff
@@ -1121,6 +1121,40 @@ function populateDatabase($oDB)
             'attributes' => "text NULL",
         ], $options);
 
+        // language tables: sourcemessage + message and constraint
+        $oDB->createCommand()->createTable(
+            '{{sourcemessage}}',
+            [
+                'id' => "pk",
+                'category' => "string(35)",
+                'message' => "text",
+            ],
+            $options
+        );
+        $oDB->createCommand()->createTable(
+            '{{message}}',
+            [
+                'id' => "integer NOT NULL",
+                'language' => "string(16)",
+                'translation' => "text",
+            ],
+            $options
+        );
+        $oDB->createCommand()->addPrimaryKey(
+            '{{message_pk}}',
+            '{{message}}',
+            ['id', 'language']
+        );
+        $oDB->createCommand()->addForeignKey(
```

Comment:
Normally we try to avoid foreign keys, because we never know how they work across MSSQL, Postgres and Maria/MySQL.

Reply:
See : https://www.yiiframework.com/doc/api/1.1/CDbMessageSource 

And yes : no constraint with mariadb/myisam

Reply:
Removed : use at your own risk for manual